### PR TITLE
Support predictors on ANSI day time interval type

### LINKED ISSUE #4149: [Audit][FEA][SPARK-36866] pushdown ANSI interval type filters to Parquet scan
Make sure GpuParquetScan can handle ANSI interval type filters correctly.

### LINKED ISSUE #4945: [FEA] Support filter and comparisons on ANSI day time interval type
**Is your feature request related to a problem? Please describe.**
ANSI interval types(YearMonthIntervalType and DayTimeIntervalType) can be read and writen from parquet after https://github.com/NVIDIA/spark-rapids/issues/4145
Then all the Expression and Executor nodes should also support interval types.
This issue only involves the following operators:

**Describe the solution you'd like**
- [ ] IsNull
- [ ] IsNotNull
- [ ] EqualNullSafe
- [ ] EqualTo
- [ ] GreaterThan
- [ ] GreaterThanOrEqual
- [ ] LessThan
- [ ] LessThanOrEqual
- [ ] FilterExec

### PR DESCRIPTION
**Pull request by res-life**
Fixes #4945
Closes #4149

1. Support predictors on ANSI day time interval type
IsNull, IsNotNull, EqualNullSafe, EqualTo, GreaterThan, GreaterThanOrEqual, LessThan, LessThanOrEqual, FilterExec

2. Add test for pushdown ANSI interval type.
No need to change code, only need to add a test case.
Parquet stores day-time interval as long and the statistics on day-time interval are values of max(long) and min(long).
ParquetFilters.createFilter already converts filters on day-time to filters on long.

3. Add converter from ANSI day time literal to scalar

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

**Review by revans2 — APPROVED** [review]
Just curious about the one test.

File: integration_tests/src/main/python/parquet_test.py [hunk 1]
```diff
@@ -806,4 +806,15 @@ def test_parquet_read_daytime_interval_gpu_file(spark_tmp_path):
...
+
+@pytest.mark.skipif(is_before_spark_330(), reason='DayTimeInterval is not supported before Pyspark 3.3.0')
+def test_parquet_push_down_on_interval_type(spark_tmp_path):
+    gen_list = [('_c1', DayTimeIntervalGen())]
+    data_path = spark_tmp_path + '/PARQUET_DATA'
+    with_cpu_session(lambda spark: gen_df(spark, gen_list).coalesce(1).write.parquet(data_path))
+    assert_gpu_and_cpu_are_equal_sql(
+        lambda spark: spark.read.parquet(data_path),
+        "testData",
+        "select * from testData where _c1 > interval '10 0:0:0' day to second")
```
[the remark this answers is not in the document]
Comment (res-life) [reply integration_tests/src/main/python/parquet_test.py:820]: I've verified by debuging. Validating pushdowns with test cases is somewhat difficult.
`interval '10 0:0:0' day to second` is transformed to `864000000000`, see below `GpuFilter` in the `Physical Plan`
For the `GpuFileGpuScan` node, the filter remains to be the original one (PT240H duration), but our code actually converted the filter, see code:

https://github.com/NVIDIA/spark-rapids/blob/branch-22.04/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScanBase.scala#L350
```
filters.flatMap(parquetFilters.createFilter).reduceOption(FilterApi.and)
```


```
== Physical Plan ==
GpuColumnarToRow false
+- GpuFilter (gpuisnotnull(c1#3) AND (c1#3 > 864000000000)), true
   +- GpuFileGpuScan parquet [c1#3] Batched: true, DataFilters: [isnotnull(c1#3), (c1#3 > INTERVAL '10 00:00:00' DAY TO SECOND)], Format: Parquet, Location: InMemoryFileIndex[file:/tmp/my-tmp/my-parquet], PartitionFilters: [], PushedFilters: [IsNotNull(c1), GreaterThan(c1,PT240H)], ReadSchema: struct<c1:interval day to second>
```

File: sql-plugin/src/main/301until330-all/scala/com/nvidia/spark/rapids/shims/GpuTypeShims.scala [hunk 2]
```diff
@@ -46,4 +46,14 @@ object GpuTypeShims {
    * @return the cuDF type if the Shim supports
    */
   def toRapidsOrNull(t: DataType): DType = null
+
+  /**
+   * Whether this Shim supports convert this type to GPU Scalar
+   * @param t
```
[the remark this answers is not in the document]
Comment (res-life) [reply sql-plugin/src/main/301until330-all/scala/com/nvidia/spark/rapids/shims/GpuTypeShims.scala:68]: done.

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

**Pull request merged**